crates/accelerate/src/high_level_synthesis.rs

```diff
@@ -907,6 +942,17 @@ fn py_run_on_dag(
 ) -> PyResult<Option<DAGCircuit>> {
     // Fast-path: check if HighLevelSynthesis can be skipped altogether. This is only
     // done at the top-level since this does not track the qubit states.
+
+    // First, we apply a super-fast (but incomplete) check to see if all the operations
+    // present in the circuit are suported by the target / are in the basis. This check
+    // is based only on the names of the operations in the circuit.
+    let op_names: HashSet<String> = dag.count_ops(py, false)?.keys().cloned().collect();
```

Comment:
Would it be possible to directly iterate over the `Keys` object instead of cloning and collecting it? The Keys should already be unique 🤔

Reply:
Done in [d25aaa0](https://github.com/Qiskit/qiskit/pull/13813/commits/d25aaa022e465d5e39852e26eedf1fec53ad0daf). 

Answering the offline question: when we call
```
let ops = dag.count_ops(py, false)?;
let mut op_keys = ops.keys();
```
and later call
```
op_keys.all(|name| borrowed_data.device_insts.contains(name))
```
the `op_keys` need to be mutable because (if I understand this correctly) while the iterator does not change the elements, it needs to be consumed mutably because the `all` methods needs to adjust its internal state.

Reply:
Thanks for the explanation!